### PR TITLE
profile page

### PR DESCRIPTION
**Pull request by davidvargaspuga**
- Authentication servlet redirects back to the same profile page of a certain user_id and returns the user email if logged in. 
- ProfileInfo servlet returns name as the first half of email.
- Profile object uses generics.

*Needs center formatting for thumbs, and deletion of the user's profile pic since we don't have access to that. *

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/ProfileInfoServlet.java [hunk 1]
```diff
@@ -33,18 +34,29 @@
...
-    HashMap<String, String> contactInfo = new HashMap<String, String>();
-    contactInfo.put("Mobile", "(123) 456-7809");
-    contactInfo.put("Email", "johndoe@gmail.com");
-    Profile profile1 = new Profile(name, projectHistory, reviews, contactInfo);
+    
+    // parse email for name
+    String email = request.getParameter("user_id");
+    int endIndex;
+    String name;
+    if(email != null && email.contains("@")){
```
Comment (DIEGOTEC18) [inline src/main/java/com/google/sps/servlets/ProfileInfoServlet.java:42]: Great way to get the username.

File: src/main/java/com/google/sps/servlets/ProfileInfoServlet.java [hunk 2]
```diff
@@ -33,18 +34,29 @@
...
+        endIndex = email.indexOf("@");
+        name = email.substring(0, endIndex);
+        System.out.println("inside");
+    }
+    else{
+        name = "John Doe";
```
Comment (DIEGOTEC18) [inline src/main/java/com/google/sps/servlets/ProfileInfoServlet.java]: I know this is a generic name but maybe using "No Username" is less misleading.

File: src/main/java/com/google/sps/servlets/ProfileProjectsServlet.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        Iterable<Entity> results = datastore.prepare(query).asIterable();
+
+        List <SearchProject> projects = new ArrayList<>();
+        for (Entity entity : results) {
+            String author = (String) entity.getProperty("author");
+
+            // only add project if author matches the email
+            if(email.equals(author)){
+
+                //Get the properties of the entity:
```
Comment (DIEGOTEC18) [inline src/main/java/com/google/sps/servlets/ProfileProjectsServlet.java:40]: Mannie changed the project property names. Remember to update them. You can see the new ones implemented in my latest Pull request.

File: src/main/webapp/profile.js [hunk 4]
```diff
@@ -13,127 +13,161 @@
...
-        console.log(profile.contactInfo.Email + "<br/>" + profile.contactInfo.Mobile);
-        contactInfoElement.display = "none";
-
-        for (var i = 0; i < profile.reviews.length; i++) {
-            const reviewElement = document.createElement('li');
-            reviewElement.innerText = profile.reviews[i];
-            console.log(profile.reviews[i]);
-            reviewElement.className = 'horiz-center';
-            reviewListElement.appendChild(reviewElement);
+async function loadProfile() {
```
Comment (DIEGOTEC18) [inline src/main/webapp/profile.js:37]: This is cool!

File: src/main/webapp/profile.js [hunk 5]
```diff
@@ -13,127 +13,161 @@
 // limitations under the License.
 
 
+// Author: Diego V.A.
```
Comment (DIEGOTEC18) [inline src/main/webapp/profile.js:16]: ;)

**Review by DIEGOTEC18 — APPROVED** [review]
Looks good to me!
Just remember to update the project's property names to match the latest ones.

**Comment by DIEGOTEC18** [comment]
image
This are the latest project's property names.

File: src/main/java/com/google/sps/servlets/ProfileInfoServlet.java (hunk 2 above)
Comment (davidvargaspuga) [reply src/main/java/com/google/sps/servlets/ProfileInfoServlet.java]: good suggestion!

File: src/main/java/com/google/sps/servlets/ProfileProjectsServlet.java (hunk 3 above)
Comment (davidvargaspuga) [reply src/main/java/com/google/sps/servlets/ProfileProjectsServlet.java:40]: updated them, thank you.

File: src/main/java/com/google/sps/servlets/ProfileInfoServlet.java (hunk 2 above)
Comment (marcelpuyat) [reply src/main/java/com/google/sps/servlets/ProfileInfoServlet.java]: +1

File: src/main/java/com/google/sps/servlets/ProfileProjectsServlet.java (hunk 3 above)
Comment (marcelpuyat) [reply src/main/java/com/google/sps/servlets/ProfileProjectsServlet.java:40]: The best way to do this (you odn't have to do it here if you don't want to) is to have a constants file that defines all these strings there. This way there's no way to have discrepancies in the names used in different files.

**Review by marcelpuyat — APPROVED** [review]
Very well done, David!

**Pull request merged**